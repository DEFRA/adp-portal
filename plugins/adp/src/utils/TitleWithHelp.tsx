import { Button } from '@material-ui/core';
import Help from '@mui/icons-material/Help';
import type { PropsWithChildren } from 'react';
import React from 'react';

export type TitleWithHelpProps = PropsWithChildren<
  Readonly<{
    href: string;
  }>
>;

export function TitleWithHelp({ children, href }: TitleWithHelpProps) {
  return (
    <div style={{ display: 'flex', flexDirection: 'row', width: '100%' }}>
      <h3 style={{ flexGrow: 1, margin: 0 }}>{children}</h3>
      <Button
        style={{ flexGrow: 0 }}
        color="primary"
        size="large"
        href={href}
        target="_blank"
        startIcon={<Help />}
      >
        Help
      </Button>
    </div>
  );
}
