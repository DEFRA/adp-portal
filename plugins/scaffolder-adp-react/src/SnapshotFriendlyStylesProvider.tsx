import React, { type PropsWithChildren } from 'react';

import { StylesProvider } from '@material-ui/core';

export function SnapshotFriendlyStylesProvider({
  children,
}: Readonly<PropsWithChildren<{}>>) {
  if (process.env.JEST_WORKER_ID === undefined)
    throw new Error('SnapshotFriendlyStylesProvider is a test-only component');
  return (
    <StylesProvider
      generateClassName={(rule, styleSheet) =>
        `${styleSheet?.options.classNamePrefix}-${rule.key}`
      }
    >
      {children}
    </StylesProvider>
  );
}
