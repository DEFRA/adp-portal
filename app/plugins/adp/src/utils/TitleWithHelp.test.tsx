import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { TitleWithHelp } from './TitleWithHelp';
import { SnapshotFriendlyStylesProvider } from './SnapshotFriendlyStylesProvider';

describe('TitleWithHelp', () => {
  it('Should render correctly', async () => {
    // arrange

    // act
    const result = render(
      <SnapshotFriendlyStylesProvider>
        <TitleWithHelp href="https://test.com">My title</TitleWithHelp>
      </SnapshotFriendlyStylesProvider>,
    );
    await waitFor(() => expect(result.baseElement).not.toBeEmptyDOMElement());

    // assert
    expect(result.baseElement).toMatchSnapshot();
  });
});
