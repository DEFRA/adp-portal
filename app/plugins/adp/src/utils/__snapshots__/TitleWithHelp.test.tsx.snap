// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithHelp Should render correctly 1`] = `
<body>
  <div>
    <div
      style="display: flex; flex-direction: row; width: 100%;"
    >
      <h3
        style="flex-grow: 1; margin: 0px;"
      >
        My title
      </h3>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
        href="https://test.com"
        style="flex-grow: 0;"
        tabindex="0"
        target="_blank"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeLarge"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HelpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 17h-2v-2h2zm2.07-7.75-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25"
              />
            </svg>
          </span>
          Help
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</body>
`;
